src/context.rs

```diff
@@ -158,6 +158,50 @@ impl Context {
         self.plan_queue.add_plan(time, Box::new(callback), phase)
     }
 
+    fn evaluate_periodic_and_schedule_next(
+        &mut self,
+        period: f64,
+        callback: impl Fn(&mut Context) + 'static,
+        phase: ExecutionPhase,
+    ) {
+        callback(self);
+        if !self.plan_queue.is_empty() {
+            let next_time = self.current_time + period;
+            self.add_plan_with_phase(
+                next_time,
+                move |context| context.evaluate_periodic_and_schedule_next(period, callback, phase),
+                phase,
+            );
+        }
+    }
+
+    /// Add a plan with specified priority to the future event list, and
+    /// continuously repeat the plan at the specified period, stopping
+    /// only once there are no other plans scheduled.
+    ///
+    /// Returns an `Id` for the newly-added plan that can be used to cancel it
+    /// if needed.
+    /// # Panics
+    ///
+    /// Panics if time is negative, infinite, or NaN.
+    pub fn add_periodic_plan_with_phase(
+        &mut self,
+        period: f64,
+        callback: impl Fn(&mut Context) + 'static,
+        phase: ExecutionPhase,
+    ) {
```

Comment:
My claim is that periodic plans should not return `Id` because that `Id` has no meaning after the first period's plan has run.